core/rawdb/accessors_chain.go

```diff
@@ -679,6 +679,15 @@ type storedReceiptRLP struct {
 	PostStateOrStatus []byte
 	CumulativeGasUsed uint64
 	Logs              []*types.Log
+
+	// Remaining fields are declared to allow the receipt RLP to be parsed without errors.
+	// However, they must not be used as they may not be populated correctly due to multiple receipt formats
+	// being combined into a single list of optional fields which can be mistaken for each other.
+	DepositNonce *uint64  `rlp:"optional"` // Added in Regolith
```

Comment:
will there be any errors if the legacy receipt field that overlaps doesn't fit in a uint64? (on mobile, can't find the legacy receipt definition)

Reply:
Turns out it does return an error. The `DepositNonce` field however can be parsed into a `big.Int` so I've modified it to omit the `DepositNonce` field - it's value now just gets parsed into `L1GasUsed`. Added a test for just the parsing code for legacy ovm receipts that uses a value greater than a uint64 can hold.